### ISSUE TITLE
WCAG 4.1.2 test 20: false positives

### ISSUE THREAD

**Issue by jzuidweg** [veraPDF/veraPDF-wcag-algs#103]
[This document](https://github.com/veraPDF/veraPDF-wcag-algs/files/7476488/Nota.De.Tweede.Helft.-.Ontwikkelingen.sportaccommodaties.2021-2030.pdf) fails **WCAG 4.1.2 test 20** ("_Caption is incorrectly tagged as a paragraph_"). However, many occurrences of this error point to text that is obviously a paragraph and not a caption.

The problem seems to be that the algorithm points to almost **any** text just before or after a figure. My suggestion would be to trigger this algorithm only when text just before or after a figure is **no longer than one or two lines long**. Optionally you could also check if the text starts with the words "**Figure**", possibly followed by a number or a composed number such as "1.2"

**Comment by bdoubrov** [comment]
We can decrease the probability depending on the number of lines in text next to the figure. Captions consisting of multiple lines are rare. 

In addition, we increase the probability, if this text starts with Figure (or its equivalents on other languages). So, if there is a multiline text next to the figure that does start with "Figure", it will still be recognized as a caption.

**Comment by jzuidweg** [comment]
I've tested **WCAG 4.1.2 test 20** ("_Caption is incorrectly tagged as a paragraph_") and it behaves correctly, so this issue appears to be solved.

However, I did notice that **WCAG 4.1.2 test 15** ("Paragraph is incorrectly tagged as a numbered heading") still gives a false positive here on page 34 ("4.2.1 Binnensportaccomodatie" which is clearly a heading).

I will close this issue and make a new issue for **WCAG 4.1.2 test 15**.
